examples/alert_triage_agent/src/aiq_alert_triage_agent/host_performance_check_tool.py

```diff
@@ -0,0 +1,172 @@
+    input_data = "\n".join(stdout_lines) if stdout_lines else ""
+
+    prompt = ToolReasoningLayerPrompts.HOST_PERFORMANCE_CHECK.format(input_data=input_data)
+
+    response = await utils.llm_ainvoke(config, builder, user_prompt=prompt)
+
+    try:
+        structured_data = response
+    except Exception as e:
+        structured_data = f'{{"error": "Failed to parse nvda_nim response", "exception": "{str(e)}", "raw_response": "{response}"}}'
+    return structured_data
+
+
+@register_function(config_type=HostPerformanceCheckToolConfig)
+async def host_performance_check_tool(config: HostPerformanceCheckToolConfig, builder: Builder):
+    async def _arun(host_id: str) -> str:
+        is_test_mode = utils.is_test_mode()
+        utils.log_header("Host Performance Analyzer")
+
+        try:
+            if not is_test_mode:
+                # In production mode, use actual Ansible connection details
+                # TODO: Replace these placeholder values with actual connection info
+                ansible_host = "your.host.example.name"  # Input your target host
+                ansible_user = "ansible_user"  # Input your SSH user
+                ansible_port = 22  # Input your SSH port
+                ansible_private_key_path = "/path/to/private/key"  # Input path to your SSH key
+
+                # Run Ansible playbook to collect performance metrics
+                output = await _run_ansible_playbook_for_host_performance_check(
+                    config=config,
+                    builder=builder,
+                    ansible_host=ansible_host,
+                    ansible_user=ansible_user, 
+                    ansible_port=ansible_port,
+                    ansible_private_key_path=ansible_private_key_path
+                )
+            else:
+                # In test mode, load performance data from test dataset
+                df = utils.load_test_data()
+
+                # Get CPU metrics from test data, falling back to static data if needed
+                data_top_cpu = utils.load_column_or_static(
+                    df=df,
+                    host_id=host_id,
+                    column="host_performance_check_tool:top_output"
+                )
+                data_ps_cpu = utils.load_column_or_static(
+                    df=df,
+                    host_id=host_id,
+                    column="host_performance_check_tool:ps_output"
+                )
+
+                output = f"`top` :{data_top_cpu} and `ps` :{data_ps_cpu}"
+
+            # Additional LLM reasoning layer on playbook output to provide a summary of the results
+            utils.log_header("LLM Reasoning", dash_length=50)
+            
+            prompt_template = f"""You are analyzing system metrics to assess CPU and memory usage. Use the output below to determine whether CPU or memory usage is abnormally high, identify which processes are consuming the most resources, and assess whether the usage patterns could explain a recent alert.
```

Comment:
I would define this as a global outside the function.

Reply:
Thanks for the catch! I missed this one when moving all the prompts to prompts.py. I'll move it over.